code/game/turfs/space/space.dm

```diff
@@ -69,7 +69,7 @@
 
 /turf/space/Entered(atom/movable/A as mob|obj)
 	..()
-	if ((!(A) || src != A.loc))
+	if ((!(A) || !(src in A.locs)))

```

Comment:
If you push a closet through a Z-level, won't this cause it to dump everything it's carrying